client/dashboard/domains/dns/add.tsx

```diff
@@ -0,0 +1,61 @@
+import { useMutation } from '@tanstack/react-query';
```

Comment:
This was refactored from the original `index` file. I split the "add" and "edit" functionalities into separate files to make the code more readable and maintainable.